cmake/modules/extensions.cmake

```diff
@@ -3095,10 +3095,10 @@ function(zephyr_string)
   endif()
 
   if(ZEPHYR_STRING_ESCAPE)
-    # If a single '\' is discovered, such as 'foo\bar', then it must be escaped like: 'foo\\bar'
-    # \\1 and \\2 are keeping the match patterns, the \\\\ --> \\ meaning an escaped '\',
-    # which then becomes a single '\' in the final string.
-    string(REGEX REPLACE "([^\\][\\])([^\\\"])" "\\1\\\\\\2" work_string "${ZEPHYR_STRING_UNPARSED_ARGUMENTS}")
+    # Escape every instance of '\' or '"' in the string. "\\1" is the matched
+    # character. "\\\\" is seen by CMake as \\, meaning an escaped '\', which
+    # then becomes a single '\' in the final string.
+    string(REGEX REPLACE "([\"\\])" "\\\\\\1" work_string "${ZEPHYR_STRING_UNPARSED_ARGUMENTS}")
```

Comment:
would be nice if we can simplify.

iirc one reason for avoiding escaping already escaped sequences was that we need to post-process the generator expanded strings, and afaict that will still be needed.

But keep up the direction :+1:

Reply:
> > we need to post-process the generator expanded strings, and afaict that will still be needed.
>
> This should no longer be perfomed. The trick is using ' for intermediate YAML strings, so the only problematic char after expansion is the single quote

True, but the `'` char must still be properly handled, also for generator expression which are expanded at generator time, and thus handle at build time. That's what I was referring to.

But I see the `'` is not properly handled, also when using genex :+1: